### PR TITLE
Add category

### PR DESCRIPTION
**Pull request by ArtemBozhok**

### CODE REVIEW

File: src/interfaces/categories-data.ts [hunk 1]
```diff
@@ -2,4 +2,12 @@ import { ICategoryItem } from './category-Item';
...
\ No newline at end of file
+}
+
+export interface CategoryTableData {
+  id: number;
+  createdAt: string;
+  updatedAt: string;
+  name: string;
+  products: number;
+};
```
Comment (rurkh) [inline src/interfaces/categories-data.ts:13]: please add new lines and othe identations

File: src/store/actions.ts [hunk 2]
```diff
@@ -1,6 +1,14 @@
+import { IActions } from "../interfaces/actions";
 import { ICategoryItem } from "../interfaces/category-Item";
-import { ADD_CATEGORY, LOAD_CATEGORIES, REQUEST_CATEGORIES } from "./types";
+import { ADD_CATEGORY, LOAD_CATEGORIES, REQUEST_ADD_CATEGORIES, REQUEST_CATEGORIES } from "./types";
```
Comment (rurkh) [inline src/store/actions.ts:3]: please, use singlequotes everywhere   except  it is pure HTML

File: src/store/actions.ts [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
 
-export const loadCategories = (categories: ICategoryItem[]) => ({type: LOAD_CATEGORIES, data: categories});
-export const fetchCategories = () => ({type: REQUEST_CATEGORIES});
-export const addCategory = (category: ICategoryItem) => ({type: ADD_CATEGORY, data: category});
\ No newline at end of file
+export const loadCategories = (categories: ICategoryItem[]): IActions => ({type: LOAD_CATEGORIES, data: categories});
+export const fetchCategories = (): IActions => ({type: REQUEST_CATEGORIES});
+export const fetchAddCategories = (name: string): IActions => ({
+  type: REQUEST_ADD_CATEGORIES,
+  data: {name},
```
Comment (rurkh) [inline src/store/actions.ts:9]: { name }, object properties require spaces, when following  common style guide

File: src/components/Forms/Category-form.tsx [hunk 4]
```diff
@@ -53,8 +44,7 @@ const CategoryForm:React.FC<CategoryFormProps> = ({dispatch, categoriesLength, h
        onSubmit={(values, { setSubmitting }) => {
          setTimeout(() => {
```
Comment (rurkh) [inline src/components/Forms/Category-form.tsx:45]: why is setTimeout needed?

File: src/components/Forms/Category-form.tsx [hunk 5]
```diff
@@ -53,8 +44,7 @@ const CategoryForm:React.FC<CategoryFormProps> = ({dispatch, categoriesLength, h
        onSubmit={(values, { setSubmitting }) => {
```
Comment (rurkh) [inline src/components/Forms/Category-form.tsx:44]: its a bad  practice to define anononymous functions inside render of components, cause it  creates new functions each render, define something like onSubmitting in Component itself.

Also i recommend, to use withFormik hook, against Formi component

File: src/store/sagas/services/category.service.ts [hunk 6]
```diff
@@ -1,6 +1,12 @@
...
 
 export async function fetchedCategories () {
-   const response = await fetch(`${root}/category`);
-   return await response.json();
+   const categories = await api.categories.get();
+   return categories.data;
+}
+
+export async function addCategories (data: IActions) {
+   const newCategory =  await api.categories.add(data);
```
Comment (rurkh) [inline src/store/sagas/services/category.service.ts:10]: what is with error handling?

**Review by rurkh — CHANGES_REQUESTED** [review]

File: src/interfaces/categories-data.ts (hunk 1 above)
Comment (ArtemBozhok) [reply src/interfaces/categories-data.ts:13]: +

File: src/store/actions.ts (hunk 2 above)
Comment (ArtemBozhok) [reply src/store/actions.ts:3]: +

File: src/store/actions.ts (hunk 3 above)
Comment (ArtemBozhok) [reply src/store/actions.ts:9]: +

File: src/components/Forms/Category-form.tsx (hunk 4 above)
Comment (ArtemBozhok) [reply src/components/Forms/Category-form.tsx:45]: Not nedded. It's has been made for emulation slow network request.

File: src/components/Forms/Category-form.tsx (hunk 5 above)
Comment (ArtemBozhok) [reply src/components/Forms/Category-form.tsx:44]: Rewrite code with HOK WithFormik.

File: src/store/sagas/services/category.service.ts (hunk 6 above)
Comment (ArtemBozhok) [reply src/store/sagas/services/category.service.ts:10]: Error handling is aded.

**Review by rurkh — APPROVED** [review]

**Pull request merged**